src/common/observer.h

```diff
@@ -20,11 +20,11 @@
 #define OBSERVER_PRINT LOG(INFO)
 #define OBSERVER_ENDL ""
 #define OBSERVER_NEWLINE ""
-#else
+#else  // defined(XGBOOST_STRICT_R_MODE) && XGBOOST_STRICT_R_MODE == 1
```

Comment:
You can remove any use of this class (currently only called in learner), and don't define the class when compiling on R.

Reply:
That would disable debug logging entirely for the R package. Not sure if we should do that.